core/src/test/java/org/apache/calcite/test/TableInRootSchemaTest.java

```diff
@@ -146,4 +175,17 @@ protected RowTable() {
     statement.close();
     connection.close();
   }
+
+  /** Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-6764">[CALCITE-6764]
+   * Field access from a nullable ROW should be nullable</a>. */
```

Comment:
I don't see where we are actually testing the result being `NULL`?

Reply:
this is implicitly tested by the Validator, which crashes without this fix.
It fails with the same error as described in the JIRA issue: `java.lang.RuntimeException: java.lang.AssertionError: Conversion to relational algebra failed to preserve datatypes`.

Reply:
Let's add it as a comment for who might overlook that the way I did

Reply:
Can you break the line after 'for', as is usual for a 'Test case for ...' javadoc comment.

Reply:
there are lots of examples with either structure, but I have added a newline